opendbc/car/hyundai/carcontroller.py

```diff
@@ -3,206 +3,154 @@
 from opendbc.car import Bus, DT_CTRL, apply_driver_steer_torque_limits, common_fault_avoidance, make_tester_present_msg, structs
 from opendbc.car.common.conversions import Conversions as CV
 from opendbc.car.hyundai import hyundaicanfd, hyundaican
-from opendbc.car.hyundai.carstate import CarState
 from opendbc.car.hyundai.hyundaicanfd import CanBus
 from opendbc.car.hyundai.values import HyundaiFlags, Buttons, CarControllerParams, CAR
 from opendbc.car.interfaces import CarControllerBase
 
 VisualAlert = structs.CarControl.HUDControl.VisualAlert
 LongCtrlState = structs.CarControl.Actuators.LongControlState
 
-# EPS faults if you apply torque while the steering angle is above 90 degrees for more than 1 second
-# All slightly below EPS thresholds to avoid fault
+# EPS faults if you apply torque while the steering angle is above 90 degrees for more than 1 second, all slightly below EPS thresholds to avoid fault
 MAX_ANGLE = 85
 MAX_ANGLE_FRAMES = 89
 MAX_ANGLE_CONSECUTIVE_FRAMES = 2
 
 
-def process_hud_alert(enabled, fingerprint, hud_control):
-  sys_warning = (hud_control.visualAlert in (VisualAlert.steerRequired, VisualAlert.ldw))
-
-  # initialize to no line visible
-  # TODO: this is not accurate for all cars
-  sys_state = 1
-  if hud_control.leftLaneVisible and hud_control.rightLaneVisible or sys_warning:  # HUD alert only display when LKAS status is active
-    sys_state = 3 if enabled or sys_warning else 4
-  elif hud_control.leftLaneVisible:
-    sys_state = 5
-  elif hud_control.rightLaneVisible:
-    sys_state = 6
-
-  # initialize to no warnings
-  left_lane_warning = 0
-  right_lane_warning = 0
-  if hud_control.leftLaneDepart:
-    left_lane_warning = 1 if fingerprint in (CAR.GENESIS_G90, CAR.GENESIS_G80) else 2
-  if hud_control.rightLaneDepart:
-    right_lane_warning = 1 if fingerprint in (CAR.GENESIS_G90, CAR.GENESIS_G80) else 2
-
-  return sys_warning, sys_state, left_lane_warning, right_lane_warning
-
-
 class CarController(CarControllerBase):
   def __init__(self, dbc_names, CP):
     super().__init__(dbc_names, CP)
     self.CAN = CanBus(CP)
     self.params = CarControllerParams(CP)
     self.packer = CANPacker(dbc_names[Bus.pt])
     self.angle_limit_counter = 0
-
     self.accel_last = 0
     self.apply_torque_last = 0
     self.car_fingerprint = CP.carFingerprint
     self.last_button_frame = 0
 
+
   def update(self, CC, CS, now_nanos):
-    actuators = CC.actuators
-    hud_control = CC.hudControl
+    if self.CP.flags & HyundaiFlags.CANFD:
+      return self.update_canfd(CC, CS, now_nanos)
+
+    (actuators, hud_control, apply_torque, apply_steer_req, torque_fault, accel, stopping, set_speed_in_units, sys_warning, sys_state, left_lane_warning,
+     right_lane_warning, tester_present_msgs) = self.update_common(CC, CS)
```

Comment:
Using a `namedtuple` may be more readable. Example of how Honda's doing it now:

https://github.com/commaai/opendbc/blob/75d4e0eb5c8fa961a665ec020eb816a50eed231c/opendbc/car/honda/carcontroller.py#L94-L96

Reply:
just https://github.com/commaai/opendbc/pull/1936/commits/9a292091888e702df70725f6916110c08e76c0cb or all the way like I did?